src/contexts/reducer.js

```diff
@@ -0,0 +1,23 @@
+import ENUMS from '../utils/enums';
+
+const { SET_CART, SET_WISHLIST, SET_PRODUCTS } = ENUMS;
+
+const reducer = (state, action) => {
+    const { type, data } = action;
+
+    switch (type) {
+        case SET_CART:
+            return { ...state, cart: data };
```

Comment:
useReducer has convetion to have action with type and payload properties. You can change data to payload to avoid confusion for others.